platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -314,4 +247,103 @@ private static boolean solrCoreExists(SolrClient client, String coreName)
+  }
+
+  @VisibleForTesting
+  SolrClient createSolrHttpClient(String url, String coreName, String coreUrl)
+      throws IOException, SolrServerException {
+    final HttpClientBuilder builder = createHttpBuilder();
+
+    createSolrCore(url, coreName, null, builder.build());
+    try (final Closer closer = new Closer()) {
+      final HttpSolrClient noRetryClient =
+          closer.with(new HttpSolrClient.Builder(coreUrl).withHttpClient(builder.build()).build());
+      final HttpSolrClient retryClient =
+          closer.with(
+              new HttpSolrClient.Builder(coreUrl)
+                  .withHttpClient(
+                      builder.setRetryHandler(new SolrHttpRequestRetryHandler(coreName)).build())
+                  .build());
+
+      return closer.returning(new PingAwareSolrClientProxy(retryClient, noRetryClient));
+    }
+  }
+
+  public boolean useTls() {
+    return StringUtils.startsWithIgnoreCase(
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
+        "https");
+  }
+
+  private HttpClientBuilder createHttpBuilder() {
+
+    HttpClientBuilder httpClientBuilder =
+        HttpClients.custom()
+            .setDefaultCookieStore(new BasicCookieStore())
+            .setMaxConnTotal(128)
+            .setMaxConnPerRoute(32);
+
+    if (useTls()) {
+      httpClientBuilder.setSSLSocketFactory(
+          new SSLConnectionSocketFactory(
+              getSslContext(),
+              DEFAULT_PROTOCOLS,
+              DEFAULT_CIPHER_SUITES,
+              SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER));
+    }
+
+    boolean useBasicAuth =
+        AccessController.doPrivileged(
+            (PrivilegedAction<Boolean>)
+                () -> Boolean.valueOf(System.getProperty("solr.useBasicAuth")));
+    if (useBasicAuth) {
+      // TODO: When this JAR becomes a bundle, convert SolrPasswordUpdate to become a bean.
+      solrPasswordUpdate.start();
+
+      httpClientBuilder.setDefaultCredentialsProvider(getCredentialsProvider());
+      httpClientBuilder.addInterceptorFirst(new PreemptiveAuth(new BasicScheme()));
+    }
+    return httpClientBuilder;
+  }
+
+  private CredentialsProvider getCredentialsProvider() {
+    CredentialsProvider provider = new BasicCredentialsProvider();
+
+    org.apache.http.auth.UsernamePasswordCredentials credentials =
+        new org.apache.http.auth.UsernamePasswordCredentials(
+            AccessController.doPrivileged(
+                (PrivilegedAction<String>) () -> System.getProperty("solr.username")),
+            getPlainTextSolrPassword());
+    provider.setCredentials(AuthScope.ANY, credentials);
+    return provider;
+  }
+
+  private String getPlainTextSolrPassword() {
+    return encryptionService.decryptValue(
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) () -> System.getProperty("solr.password")));
```

Comment:
Does the system property password need to be encrypted for this to work? The default is not so I guess the encryption service is smart enough. Just wanted to make sure.

Reply:
That is correct. It is covered in the documentation, in the file `hardening-solr.adoc`. I will add some more clarification about this point to the docs in the next push. 

>Note that the default password is not enclosed in `ENC()` because that is not necessary for clear text.